### ISSUE TITLE
Train accuracy converge early at ~ 0.9 during pretraining with custom aido_dna_debug backbone

### ISSUE THREAD

**Issue by CrazyHsu** [genbio-ai/ModelGenerator#14]
Hi, thank you for developing the ModelGenerator tool — it's been a great resource. 
Recently, I’ve been trying to use it for pretraining a model from scratch on my own dataset.

Here's the command I tested:
```
mgen fit \
  --model MLM \
  --data MLMDataModule \
  --data.path /path/to/my_DNA_data \
  --trainer.max_epochs 5 \
  --model.backbone aido_dna_debug \
  --trainer.strategy ddp_find_unused_parameters_true \
  --data.init_args.num_workers=4 \
  --data.init_args.batch_size=512
```
I redefined the `aido_dna_debug` backbone as follows:
```
class aido_dna_debug(GenBioBERT):
    """
    A small DNA dense transformer model created from scratch for debugging purposes only
    """
    def __init__(self, *args, **kwargs):
        config_overwrites = {
            "hidden_size": 512,
            "num_hidden_layers": 16,
            "num_attention_heads": 16,
            "intermediate_size": 2028,
        }
        super().__init__(
            *args, from_scratch=True, config_overwrites=config_overwrites, **kwargs
        )
```
During training, I observed that both the training loss and accuracy converged rapidly during the initial training stages, with the loss plateauing around 0.20 and the accuracy remaining at approximately 0.9 like below. This is unexpected, as I would typically expect the accuracy to gradually approach 1.0 with continued training.
![Image](https://github.com/user-attachments/assets/3e33ea95-62cb-4512-acf8-8d686a4bcedd)
And I also observed the same behavior when switching `--data.path` to public datasets like:
- `genbio-ai/100M-random-promoters`
- `leannmlindsey/GUE`

Is this expected behavior? Or is there something I'm missing regarding the model, data module, or training setup?
Any help would be greatly appreciated!

Feng

**Comment by cnellington** [comment]
Hi @CrazyHsu, this is the expected behavior. MLM will give ~90% when randomly guessing on masked tokens. 
- We get 85% accuracy from copying the input exactly (15% masking, no token corruption)
- We get 88.75% from randomly guessing masked values ATCG (25% success on the 15% of masked tokens)
- +/- a bit on `leannmlindsey/GUE` because nucleotide usage is not uniformly distributed in natural DNA, but `genbio-ai/100M-random-promoters` should give close to 88.75 acc because these sequences are truly random.

The task and data seem to work correctly though. If I limit `leannmlindsay/GUE` to a single sample and add "position_embedding_type": "absolute" to your `config_overwrites`, the model will quickly learn to memorize this sequence and converge to 100% accuracy. The default position embedding type is "rope" though, which makes it quite hard for the model to memorize a single output since all outputs are based on relative positioning.

It is always good to train a very small model for MLM to see what the baseline accuracy should be for random guessing while accounting for token usage biases. Anything above this means the model is reconstructing the data better than random.
